src/uct/ugni/base/ugni_iface.c

```diff
@@ -234,136 +232,54 @@ static ucs_status_t uct_ugni_fetch_pmi()
-                            modes, cdm_handle);
-    if (GNI_RC_SUCCESS != ugni_rc) {
-        ucs_error("GNI_CdmCreate failed, Error status: %s %d",
-                  gni_err_str[ugni_rc], ugni_rc);
-        return UCS_ERR_NO_DEVICE;
-    }
-
-    /* For now we use the first device for allocation of the domain */
-    ugni_rc = GNI_CdmAttach(*cdm_handle, job_info.devices[device_index].device_id,
-                            address, nic_handle);
-    if (GNI_RC_SUCCESS != ugni_rc) {
-        ucs_error("GNI_CdmAttach failed (domain id %d, %d), Error status: %s %d",
-                  *domain_id, ugni_domain_global_counter, gni_err_str[ugni_rc], ugni_rc);
-        return UCS_ERR_NO_DEVICE;
-    }
-
-    ++ugni_domain_global_counter;
-    return UCS_OK;
-}
-
-ucs_status_t ugni_activate_iface(uct_ugni_iface_t *iface)
-{
-    ucs_status_t status;
-    gni_return_t ugni_rc;
-    uint32_t pe_address;
-
-    if(iface->activated) {
-        return UCS_OK;
-    }
-
-    status = uct_ugni_init_nic(0, &iface->domain_id,
-                               &iface->cdm_handle, &iface->nic_handle,
-                               &pe_address);
-    if (UCS_OK != status) {
-        ucs_error("Failed to UGNI NIC, Error status: %d", status);
-        return status;
-    }
-
-    ucs_debug("Made ugni interface. iface->dev->nic_addr = %i iface->domain_id = %i", iface->dev->address, iface->domain_id);
-
-    ugni_rc = GNI_CqCreate(iface->nic_handle, UCT_UGNI_LOCAL_CQ, 0,
-                           GNI_CQ_NOBLOCK,
-                           NULL, NULL, &iface->local_cq);
-    if (GNI_RC_SUCCESS != ugni_rc) {
-        ucs_error("GNI_CqCreate failed, Error status: %s %d",
-                  gni_err_str[ugni_rc], ugni_rc);
-        return UCS_ERR_NO_DEVICE;
-    }
-    iface->activated = true;
-
-    /* iface is activated */
-    return UCS_OK;
-}
-
-ucs_status_t ugni_deactivate_iface(uct_ugni_iface_t *iface)
-{
-    gni_return_t ugni_rc;
-
-    if(!iface->activated) {
-        return UCS_OK;
-    }
-
-    ugni_rc = GNI_CqDestroy(iface->local_cq);
-    if (GNI_RC_SUCCESS != ugni_rc) {
-        ucs_warn("GNI_CqDestroy failed, Error status: %s %d",
-                 gni_err_str[ugni_rc], ugni_rc);
-        return UCS_ERR_IO_ERROR;
-    }
-    ugni_rc = GNI_CdmDestroy(iface->cdm_handle);
-    if (GNI_RC_SUCCESS != ugni_rc) {
-        ucs_warn("GNI_CdmDestroy error status: %s (%d)",
-                 gni_err_str[ugni_rc], ugni_rc);
-        return UCS_ERR_IO_ERROR;
-    }
-
-    iface->activated = false ;
-    return UCS_OK;
-}
-
 static ucs_mpool_ops_t uct_ugni_flush_mpool_ops = {
     .chunk_alloc   = ucs_mpool_chunk_malloc,
     .chunk_release = ucs_mpool_chunk_free,
     .obj_init      = NULL,
     .obj_cleanup   = NULL
 };
 
+void uct_ugni_cleanup_base_iface(uct_ugni_iface_t *iface)
+{
+    ucs_arbiter_cleanup(&iface->arbiter);
+    ucs_mpool_cleanup(&iface->flush_pool, 1);
+    GNI_CqDestroy(iface->local_cq);
```

Comment:
@shamisp Thing is, the next PR pulls out all of the GNI_Cq* calls and has common spin locked code paths with error logging. I'd like to fix these problems in the next PR instead.